packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart

```diff
@@ -116,10 +117,38 @@ of the Android app that is using this module.''',
     final shorebirdYaml = shorebirdEnv.getShorebirdYaml()!;
     final appId = shorebirdYaml.getAppId();
     final app = await codePushClientWrapper.getApp(appId: appId);
+    final release = await codePushClientWrapper.getRelease(
+      appId: appId,
+      releaseVersion: releaseVersion,
+    );
+    final shorebirdFlutterRevision = shorebirdEnv.flutterRevision;
+    if (release.flutterRevision != shorebirdFlutterRevision) {
+      final installFlutterRevisionProgress = logger.progress(
+        'Switching to Flutter revision ${release.flutterRevision}',
+      );
+      try {
+        await shorebirdFlutterManager.installRevision(
+          revision: release.flutterRevision,
+        );
+        installFlutterRevisionProgress.complete();
+      } catch (error) {
+        installFlutterRevisionProgress.fail('$error');
+        return ExitCode.software.code;
+      }
+    }
 
     final buildProgress = logger.progress('Building patch');
     try {
-      await buildAar(buildNumber: buildNumber);
+      await runScoped(
+        () => buildAar(buildNumber: buildNumber),
+        values: {
+          shorebirdEnvRef.overrideWith(
+            () => ShorebirdEnv(
+              flutterRevisionOverride: release.flutterRevision,
+            ),
+          )
+        },
+      );
```

Comment:
Eventually this can be abstracted by an `AarBuilder` once we clean up the `ShorebirdBuildMixin`